scanner/Makefile

```diff
@@ -24,7 +24,7 @@ ifeq ($(shell uname -s),Darwin)
 	HOST_OS := darwin
 endif
 
-CGO_ENABLED := 0
+CGO_ENABLED ?= 0
```

Comment:
We already [export `CGO_ENABLE` in `make/env.mk`](https://github.com/stackrox/stackrox/blob/master/make/env.mk#L14), so I think we should to repeat ourselves:

```suggestion
```

Maybe replace the `ifeq` in `env.mk`, because the idiomatic way of doing that is with `=?`.

Reply:
Good find, thanks for it. Somehow I did not bother to check around.

With the current state of things, `CGO_ENABLED` gets set to `1` in `env.mk` which leads to a sudden discovery that all architectures but x86 fail building upstream (e.g. [arm64](https://github.com/stackrox/stackrox/actions/runs/8803083847/job/24160366550?pr=10853), others are clickable from it).

I wonder if someone **knew** these don't work but did not leave the knowledge e.g. in the Makefile where `CGO_ENABLED` was set to `0`?
I am thankful that it's one of the more lucky cases when CI helps catch the problem versus more subtle ones that don't break in CI but break in runtime.